x/emissions/keeper/queryserver/query_server_inferences_test.go

```diff
@@ -254,3 +254,155 @@ func (s *KeeperTestSuite) TestGetNetworkInferencesAtBlock() {
+					ReputerRequestNonce: reputerLossRequestNonce,
+					TopicId:             topicId,
+				},
+			},
+			{
+				ValueBundle: &types.ValueBundle{
+					Reputer:             reputer4,
+					CombinedValue:       alloraMath.MustNewDecFromString(".0000115363240547692"),
+					ReputerRequestNonce: reputerLossRequestNonce,
+					TopicId:             topicId,
+				},
+			},
+		},
+	}
+
+	err = keeper.InsertReputerLossBundlesAtBlock(s.ctx, topicId, lossBlockHeight, reputerLossBundles)
+	require.NoError(err)
+
+	// Set Stake
+	stake0, ok := cosmosMath.NewIntFromString("210535101370326000000000")
+	s.Require().True(ok)
+	stake1, ok := cosmosMath.NewIntFromString("216697093951021000000000")
+	s.Require().True(ok)
+	stake2, ok := cosmosMath.NewIntFromString("161740241803855000000000")
+	s.Require().True(ok)
+	stake3, ok := cosmosMath.NewIntFromString("394848305052250000000000")
+	s.Require().True(ok)
+	stake4, ok := cosmosMath.NewIntFromString("206169717590569000000000")
+	s.Require().True(ok)
+	err = keeper.AddStake(s.ctx, topicId, reputer0, stake0)
+	require.NoError(err)
+	err = keeper.AddStake(s.ctx, topicId, reputer1, stake1)
+	require.NoError(err)
+	err = keeper.AddStake(s.ctx, topicId, reputer2, stake2)
+	require.NoError(err)
+	err = keeper.AddStake(s.ctx, topicId, reputer3, stake3)
+	require.NoError(err)
+	err = keeper.AddStake(s.ctx, topicId, reputer4, stake4)
+	require.NoError(err)
+
+	s.ctx = s.ctx.WithBlockHeight(inferenceBlockHeight)
+
+	// Set Inferences
+	inferences := types.Inferences{
+		Inferences: []*types.Inference{
+			{
+				Inferer:     reputer0,
+				Value:       alloraMath.MustNewDecFromString("-0.035995138925040600"),
+				TopicId:     topicId,
+				BlockHeight: inferenceBlockHeight,
+			},
+			{
+				Inferer:     reputer1,
+				Value:       alloraMath.MustNewDecFromString("-0.07333303938740420"),
+				TopicId:     topicId,
+				BlockHeight: inferenceBlockHeight,
+			},
+			{
+				Inferer:     reputer2,
+				Value:       alloraMath.MustNewDecFromString("-0.1495482917094790"),
+				TopicId:     topicId,
+				BlockHeight: inferenceBlockHeight,
+			},
+			{
+				Inferer:     reputer3,
+				Value:       alloraMath.MustNewDecFromString("-0.12952123274063800"),
+				TopicId:     topicId,
+				BlockHeight: inferenceBlockHeight,
+			},
+			{
+				Inferer:     reputer4,
+				Value:       alloraMath.MustNewDecFromString("-0.0703055329498285"),
+				TopicId:     topicId,
+				BlockHeight: inferenceBlockHeight,
+			},
+		},
+	}
+
+	err = keeper.InsertInferences(s.ctx, topicId, inferenceNonce, inferences)
+	s.Require().NoError(err)
+
+	// Update epoch topic epoch last ended
+	err = keeper.UpdateTopicEpochLastEnded(s.ctx, topicId, inferenceBlockHeight)
+	require.NoError(err)
+
+	// Test querying the server
+	req := &types.QueryLatestNetworkInferencesAtBlockRequest{
+		TopicId: topicId,
+	}
+	response, err := queryServer.GetLatestNetworkInference(s.ctx, req)
+	require.NotNil(response, "Response should not be nil")
```

Comment:
It would be good to at least add some length / not nil checks here. E.g In this case with 5 inferers: `len(response.InfererWeights) == 5`

Reply:
yup was planning to do so before merging, thanks for noticing!